pkg/distributor/distributor.go

```diff
@@ -276,8 +329,14 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 		}
 		stream.Entries = stream.Entries[:n]
 
-		keys = append(keys, util.TokenFor(userID, stream.Labels))
-		streams = append(streams, streamTracker{stream: stream})
+		if d.cfg.ShardStreams.Mode != "never" {
+			derivedKeys, derivedStreams := shardStream(stream, d.cfg, d.streamSharder, userID)
+			keys = append(keys, derivedKeys...)
+			streams = append(streams, derivedStreams...)
+		} else {
+			keys = append(keys, util.TokenFor(userID, stream.Labels))
+			streams = append(streams, streamTracker{stream: stream})
+		}
```

Comment:
Very minor non-blocking nit which speaks to style:

Try avoid double-negatives, as it helps when reading the code (especially when it's 2AM and you've been paged and you don't know the code).

```suggestion
		if d.cfg.ShardStreams.Mode == "never" {
			keys = append(keys, util.TokenFor(userID, stream.Labels))
			streams = append(streams, streamTracker{stream: stream})
		} else {
			derivedKeys, derivedStreams := shardStream(stream, d.cfg, d.streamSharder, userID)
			keys = append(keys, derivedKeys...)
			streams = append(streams, derivedStreams...)
		}
```

Reply:
Agreed, ty for catching it.